commit 48dae46676d1acb632a3e1e14d02879d57618b5d
Author: Hannes Reinecke <hare@suse.de>
Date:   Mon Jan 29 07:39:48 2024 +0100

    nvme: enable retries for authentication commands
    
    Authentication commands might trigger a lengthy computation on the
    controller or even a callout to an external entity.
    In these cases the controller might return a status without the DNR
    bit set, indicating that the command should be retried.
    This patch enables retries for authentication commands  by setting
    NVME_SUBMIT_RETRY for __nvme_submit_sync_cmd().
    
    Reported-by: Martin George <marting@netapp.com>
    Signed-off-by: Hannes Reinecke <hare@suse.de>
    Reviewed-by: Chaitanya Kulkarni <kch@nvidia.com>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
    Signed-off-by: Keith Busch <kbusch@kernel.org>

diff --git a/drivers/nvme/host/auth.c b/drivers/nvme/host/auth.c
index bf69be8966f4..a264b3ae078b 100644
--- a/drivers/nvme/host/auth.c
+++ b/drivers/nvme/host/auth.c
@@ -58,7 +58,7 @@ static int nvme_auth_submit(struct nvme_ctrl *ctrl, int qid,
 			    void *data, size_t data_len, bool auth_send)
 {
 	struct nvme_command cmd = {};
-	nvme_submit_flags_t flags = 0;
+	nvme_submit_flags_t flags = NVME_SUBMIT_RETRY;
 	struct request_queue *q = ctrl->fabrics_q;
 	int ret;